### PR TITLE
feat: add QtiTestRepository::findByDelivery

### PR DESCRIPTION
**Pull request by gabrielfs7**
Ticket: https://oat-sa.atlassian.net/browse/TR-2966

## Description

- Introduced a new repository to get a QTI test with language.
- No other data is necessary so far, so only language is added.

## TODO 

- [x] Unit tests
- [x] Update composer with packages

## PRs

- https://github.com/oat-sa/extension-tao-testqti/pull/2218
- https://github.com/oat-sa/extension-tao-bosa-selor/pull/118
- https://github.com/oat-sa/extension-tao-delivery-rdf/pull/420
- https://github.com/oat-sa/tao-core/pull/3444

### CODE REVIEW

**Review by hectoras — APPROVED** [review]

File: model/Domain/Model/QtiTest.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2021 (original work) Open Assessment Technologies SA;
+ *
+ * @author Ricardo Quintanilha <ricardo.quintanilha@taotesting.com>
```
Comment (hectoras) [inline model/Domain/Model/QtiTest.php:20]: - [ ] year++
- [ ] Author seems wrong :)

Comment (gabrielfs7) [reply model/Domain/Model/QtiTest.php]: Change year and removed the author. I am the _Zorro_ of the PRs, so nobody knows who I am (except git history 😄 )

File: test/unit/model/Infrastructure/QtiTestRepositoryTest.php [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2021 (original work) Open Assessment Technologies SA;
```
Comment (shpran) [inline test/unit/model/Infrastructure/QtiTestRepositoryTest.php:18]:
```suggestion
 * Copyright (c) 2022 (original work) Open Assessment Technologies SA;
```

**Review by shpran — APPROVED** [review]
- [x] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

File: model/Domain/Model/QtiTestRepositoryInterface.php [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2022 (original work) Open Assessment Technologies SA;
+ *
+ * @author Ricardo Quintanilha <ricardo.quintanilha@taotesting.com>
+ */
```
Comment (hectoras) [inline model/Domain/Model/QtiTestRepositoryInterface.php:21]: nitpick: Same as previously (this is a new file) :)

```suggestion
 *
 * Copyright (c) 2022 (original work) Open Assessment Technologies SA;
 */
```

**Comment by AntonNeviarouski** [comment]
QA verified

**Pull request merged**